shell/src/test/java/tachyon/shell/TFsShellTest.java

```diff
@@ -203,12 +203,56 @@ public void copyFromLocalFileToDstPathTest() throws IOException {
   }
 
   @Test
+  public void copyFromLocalOverwriteTest() throws IOException {
+    final int LEN1 = 10;
+    final int LEN2 = 20;
+    File testFile1 = generateFileContent("/testFile1", BufferUtils.getIncreasingByteArray(LEN1));
+    File testFile2 = generateFileContent("/testFile2", BufferUtils.getIncreasingByteArray(LEN2));
+    TachyonURI tachyonFilePath = new TachyonURI("/testFile");
+
+    // Write the first file
+    String[] cmd1 = {"copyFromLocal", testFile1.getPath(), tachyonFilePath.getPath()};
+    mFsShell.copyFromLocal(cmd1);
+    Assert.assertEquals(getCommandOutput(cmd1), mOutput.toString());
+    TachyonFile tFile = mTfs.getFile(tachyonFilePath);
+    Assert.assertTrue(BufferUtils.equalIncreasingByteArray(LEN1, readContent(tFile, LEN1)));
+
+    // Write the second file to the same location, which should cause an exception
+    String[] cmd2 = {"copyFromLocal", testFile2.getPath(), tachyonFilePath.getPath()};
+    try {
+      mFsShell.copyFromLocal(cmd2);
+      Assert.assertTrue(false);
+    } catch (IOException e) {}
+    // Make sure the original file is intact
+    Assert.assertTrue(BufferUtils.equalIncreasingByteArray(LEN1, readContent(tFile, LEN1)));
+  }
+
+  @Test
+  public void copyFromLocalAtomicTest() throws IOException {
+    // copyFromLocal should not leave around any empty file metadata if it fails in the middle of
+    // copying a file
+    File testFile1 = generateFileContent("/testFile1", BufferUtils.getIncreasingByteArray(10));
+    TachyonURI tachyonFilePath = new TachyonURI("/testFile");
+    // Set testFile1 to be not readable, so that when we try to open it, we fail. NOTE: for this to
+    // test anything, we depend on the implementation of copyFromLocal creating the destination file
+    // in Tachyon before it tries to open the source file
```

Comment:
That'll throw in the copyFromLocal method, when we check for the source file's existence, which is before we create the destination file, so I'm not sure that would test the atomicity we added here.